### ISSUE TITLE
Copy to clipboard doesn't work on Wayland

### ISSUE THREAD

**Issue by pintassilgo** [flameshot-org/flameshot#2848]
### Flameshot Version

Flameshot v12.1.0 (001726a5)

### Installation Type

Compiled from source

### Operating System type and version

Linux

### Description

Almost no "copy to clipboard" work on Wayland, both text or image:
- Ctrl+C or button after creating a screenshot
- "Copy URL" or "Image to Clipboard" on "Upload image" dialog (after uploading image to Imgur)

This one works, might be used as a tip to find the issue:
- "Copy URL" in "Upload History" dialog.

This is restricted to Wayland, I tried logging with X11 and it works.
But it isn't a limitation of Wayland because other apps obviously works fine to copy things to clipboard, be it image or text.

Tried with official openSUSE Tumbleweed package (12.0.0) and compiling latest source by myself, both are affected.

### Steps to reproduce

As in description.

### System Information

Operating System: openSUSE Tumbleweed 20220725
KDE Plasma Version: 5.25.3
KDE Frameworks Version: 5.96.0
Qt Version: 5.15.5
Kernel Version: 5.18.11-1-default (64-bit)
Graphics Platform: Wayland
Processors: 12 × AMD Ryzen 5 1600 Six-Core Processor
Memory: 31,3 GiB of RAM
Graphics Processor: AMD Radeon RX 570 Series

**Comment by mmahmoudian** [comment]
Have you followed https://flameshot.org/docs/guide/wayland-help/#can-t-screen-anything-on-wayland-kde ? Basically have you installed You install `xdg-desktop-portal-kde` and `xdg-desktop-portal`?

**Comment by pintassilgo** [comment]
Yes, both were already installed.

Flameshot is working perfectly here, it's just that it can't copy things to clipboard.

I mentioned above in "Descrption" the only situation in which copying to clipboard is working, I guess it might help to understand the root of the issue.

**Comment by haizaar** [comment]
Same happens on NixOS with KDE 5.23.3 and flameshot-12.1.0.
I've been using NixOS with KDE for several months and copying to keyboard using `CTRL-C` shortcut never worked.

**Comment by mmahmoudian** [comment]
> I've been using NixOS with KDE for several months and copying to keyboard using CTRL-C shortcut never worked.

Just to make sure, you have checked that you are on Wayland, and in the Flameshot config Ctrl-c is bound to copy?

**Comment by haizaar** [comment]
Yes, 100% on wayland:
```
$ echo $XDG_SESSION_TYPE
wayland
```

Here is the snipped from Flameshot config screen:

![Screenshot_20220729_210050](https://user-images.githubusercontent.com/58201/181745848-5ec12f63-a16c-4a5c-8b60-7aff42cc4d89.png)

**Comment by haizaar** [comment]
Other shortcuts like `Ctrl+S` work just fine.

When I press `Ctrl-C` it does say "Capture saved to clipboard" but it's not actually.

**Comment by pintassilgo** [comment]
> Just to make sure, you have checked that you are on Wayland, and in the Flameshot config Ctrl-c is bound to copy?

Just to be clear, the issue is not related to key pressing, because (1) every other shortcut is working and (2) it's still a issue when you try to copy to clipboard by clicking screen buttons instead of using shortcuts (with one single working exception as described in first comment).

**Comment by haizaar** [comment]
It's working from the notification pop menu (as in below), I guess it's coming from Plasma at this stage, not flameshot. 

I'd like to see if [this](https://github.com/flameshot-org/flameshot/issues/2773#issuecomment-1199181421) helps once I manage to build it on Nix

![20220729_235056](https://user-images.githubusercontent.com/58201/181775341-30eff18c-1b24-4caf-8286-35e5e53a37d0.jpg)

**Comment by pintassilgo** [comment]
> I'd like to see if [this](https://github.com/flameshot-org/flameshot/issues/2773#issuecomment-1199181421) helps once I manage to build it on Nix

Thank you, that fixed **part** of the issue.

After building with `cmake ../ -DUSE_WAYLAND_CLIPBOARD=true`, copying image now work on Wayland, both by Ctrl+C and by clicking button.

But it still not perfect because copying URL string on upload still doesn't work. A desktop notification display it was copied, but it wasn't. And if I click `Copy URL` button from "Upload image" dialog it also doesn't work:
![image](https://user-images.githubusercontent.com/5483864/181781637-e04e3070-4b2f-4837-8820-16fac9958a4e.png)

But from "Upload History" dialog, as I said in initial comment clicking in `Copy URL` **does work**.
![image](https://user-images.githubusercontent.com/5483864/181781909-9575f7ba-07b0-4e8c-bb19-5b057ccb7545.png)

So now for this last remaining issue I believe it's something fixable in Flameshot code because there's already a place in which copy URL command works.

**Comment by pintassilgo** [comment]
Is there any reason for not using `-DUSE_WAYLAND_CLIPBOARD=true` in official binaries? As I said in previous comment, it fixes part of this Wayland issue. Now I just tested on X11 and it works on there too, so for me at least there's no negative side effect on compiling with that flag.

**Comment by haizaar** [comment]
So I managed to recompile flameshot with `-DUSE_WAYLAND_CLIPBOARD=true` on NixOS and now both `Ctrl-C` and clicking button work! Hooray!

If anyone is on NixOS, you can put the following snippet into a separate file and include in your `configuration.nix` (make sure to remove `flameshot` from your package list in other places of course):
```nix
{ config, pkgs, ... }:

let 

  flameshotPin = import (pkgs.fetchFromGitHub {
    owner = "haizaar";
    repo = "nixpkgs";
    rev = "ca6081cbce02589ad9594350ce6e5bca39b09ad1";
    sha256 = "6lOu2O4KAYfGccHdZQUEr2A3mdwEtcr3S7pyLxxlZII=";
  }) {};

in {
  environment.systemPackages = [ flameshotPin.flameshot ];
}
```

**Comment by pintassilgo** [comment]
> So I managed to recompile flameshot with -DUSE_WAYLAND_CLIPBOARD=true on NixOS and now both Ctrl-C and clicking button work! Hooray!

But just image copying, right? What about auto path copying on save and auto URL copying on upload? These are still not working for me.

**Comment by haizaar** [comment]
Yes, just the image copying. Getting file path after save into clipboard still doesn't work :/

**Comment by pintassilgo** [comment]
Well, I was curious enough to fix it by myself.

As said above, recompiling with `-DUSE_WAYLAND_CLIPBOARD=true` fixes copying image. Here is the code:
https://github.com/flameshot-org/flameshot/blob/54690d5be8d8b23ff4736bda553e99de1a8bc9a7/src/utils/screenshotsaver.cpp#L109-L114

So I simply done the exact same thing for copying text, which still doesn't work with current source even with that compile flag:

[*flameshotdaemon.cpp*](https://github.com/flameshot-org/flameshot/blob/bd3431a9f5af89f013966c45933115a6bb56eaae/src/core/flameshotdaemon.cpp)
```diff
+
+#if USE_WAYLAND_CLIPBOARD
+#include <KSystemClipboard>
+#include <QMimeData>
+#endif
+
```
```diff
void FlameshotDaemon::attachTextToClipboard(QString text, QString notification)
{
    // Must send notification before clipboard modification on linux
    if (!notification.isEmpty()) {
        AbstractLogger::info() << notification;
    }

    m_hostingClipboard = true;
+
+#ifdef USE_WAYLAND_CLIPBOARD
+    auto* mimeData = new QMimeData();
+
+    mimeData->setText(text);
+    KSystemClipboard::instance()->setMimeData(mimeData,
+                                            QClipboard::Clipboard);
+#else
    QClipboard* clipboard = QApplication::clipboard();

    clipboard->blockSignals(true);
    // This variable is necessary because the signal doesn't get blocked on
    // windows for some reason
    m_clipboardSignalBlocked = true;
    clipboard->setText(text);
    clipboard->blockSignals(false);
+#endif
}
```

Recompiled and copying path and URL is now working!

It seems to be working perfectly, but I'm not sure if that's the optimal way to fix so I'm not submit a PR.

**Comment by mmahmoudian** [comment]
@borgmanJeremy what do you think about https://github.com/flameshot-org/flameshot/issues/2848#issuecomment-1199796142

**Comment by borgmanJeremy** [comment]
Yes I think that is a good change to make. Im also fine with changing the packages to include the wayland clipboard dependencies as long as someone else does it. Not all distro's support it and I hate spending time on packaging.

**Comment by 29-ux** [comment]
Enviado desde mi Huawei de Telcel.-------- Mensaje original --------De: borgmanJeremy ***@***.***>Fecha: dom., 31 de julio de 2022 9:44 a. m.Para: flameshot-org/flameshot ***@***.***>CC: Subscribed ***@***.***>Asunto: Re: [flameshot-org/flameshot] Copy to clipboard doesn't work on Wayland (Issue #2848)
Yes I think that is a good change to make. Im also fine with changing the packages to include the wayland clipboard dependencies as long as someone else does it. Not all distro's support it and I hate spending time on packaging.

**Comment by 29-ux** [comment]
Enviado desde mi Huawei de Telcel.-------- Mensaje original --------De: Adilene Chairez ***@***.***>Fecha: dom., 31 de julio de 2022 9:46 a. m.Para: flameshot-org/flameshot ***@***.***>, borgmanJeremy ***@***.***>, flameshot-org/flameshot ***@***.***>CC: Subscribed ***@***.***>Asunto: Re: [flameshot-org/flameshot] Copy to clipboard doesn't work on Wayland (Issue #2848)Enviado desde mi Huawei de Telcel.-------- Mensaje original --------De: borgmanJeremy ***@***.***>Fecha: dom., 31 de julio de 2022 9:44 a. m.Para: flameshot-org/flameshot ***@***.***>CC: Subscribed ***@***.***>Asunto: Re: [flameshot-org/flameshot] Copy to clipboard doesn't work on Wayland (Issue #2848)
Yes I think that is a good change to make. Im also fine with changing the packages to include the wayland clipboard dependencies as long as someone else does it. Not all distro's support it and I hate spending time on packaging.

**Comment by morrolinux** [comment]
Just out of curiosity: is this going to be merged then?

**Comment by mmahmoudian** [comment]
@morrolinux the situation is that we are open to this change but we need someone to step up, do it, and keep maintaining it. Are you interested to contribute regarding this matter?

**Comment by Kirottu** [comment]
On arch the official package already builds with said flag, yet it still does not work with everything else being fine except for actually copying the screenshot. I am using DWL as my compositor, and desktop portals are functioning correctly.

**Comment by BrianLeishman** [comment]
Same on Ubuntu 22.04 on Gnome/Wayland. Clicking copy here or hitting Ctrl+C seems to think it's being done successfully, but the clipboard is empty.

![image](https://user-images.githubusercontent.com/1208702/187932980-9fb7509f-6bd5-4fe4-beea-60b4dc6924b9.png)

Also possibly related, Ctrl+S does nothing at all, but the save to file icon works. The shortcuts are configured correctly.

**Comment by gerhardbeck** [comment]
I can confirm it for Zorin OS 16.1, which is basically Ubuntu 20.04 LTS with Gnome
But I think it's not the OS Problem, as it worked very well with Ubuntu 20.04 LTS itself.
Therefore I tried different installations: the snaps didn't work, the installation from flathub (all Versions 12) copies to clipboard
see also https://github.com/flameshot-org/flameshot/issues/2857#issuecomment-1242999573

@https://github.com/flameshot-org/flameshot/issues/2813

**Comment by zorn-v** [comment]
I sketched a little script to easy build and install flameshot with `USE_WAYLAND_CLIPBOARD` flag enabled on debian based distros. 
It also respects new version if it will be released.

```bash
#!/bin/bash

set -e

rm -rf build
mkdir -p build
cd build

sudo apt install devscripts equivs libkf5guiaddons-dev -y
sudo mk-build-deps -t'apt-get -y' -ir flameshot

apt source flameshot
SRC_DIR=`ls | cut -f1 | head -n1`

pushd ${SRC_DIR}
sed -i 's/-DFLAMESHOT_ABOUT_DEB_VERSION/-DUSE_WAYLAND_CLIPBOARD=1 -DFLAMESHOT_ABOUT_DEB_VERSION/' debian/rules
EMAIL=nobody@email.com dch -n ''
debuild -i -j -us -uc
popd

sudo apt purge flameshot-build-deps devscripts equivs libkf5guiaddons-dev --auto-remove -y

sudo dpkg -i flameshot_*.deb
rm -rf ../build

```

**Comment by ahangarha** [comment]
> The workaround to fix this remaining issue is launching flameshot with `GDK_BACKEND=x11`.

Didn't work for me on Kubuntu 22.10

**Comment by DominickVale** [comment]
So, since i had already invested hours in trying to even make flameshot open in hyprland, I decided to get this working no matter what.
I just found out that the image is actually copied in wl-copy, but pasting it doesn't work unless i do this:
```wl-paste | wl-copy```
which works. I have no idea why, but i hope it'll help someone in the meantime... 

I also found out that up to a certain size of the screenshot, let's say 150x150, copying straight from flameshot works, but if make a bigger one, it doesn't work anymore, until i run the aforementioned command.
As of right now I'm going to try to make a CopyQ command that does that when selecting the image in the menu

**Comment by ExistingProgrammer** [comment]
Copying to clipboard works if we run:
`flameshot gui --raw | wl-copy`

**Comment by simonbcn** [comment]
Same problem on Arch Linux and Flameshot v12.1.0 (-) compiled with Qt 5.15.5.

**Comment by y0nei** [comment]
> Copying to clipboard works if we run: `flameshot gui --raw | wl-copy`

Only this solution worked for me @ Hyprland even though i compiled with the `-DUSE_WAYLAND_CLIPBOARD=1` flag

**Comment by neoscopio** [comment]
It's still an issue with

> Operating System: Kubuntu 23.04
> KDE Plasma Version: 5.27.4
> KDE Frameworks Version: 5.104.0
> Qt Version: 5.15.8
> Kernel Version: 6.2.0-25-generic (64-bit)
> Graphics Platform: Wayland
> Processors: 4 × Intel® Core™ i5-4460 CPU @ 3.20GHz
> Memory: 7.6 GiB of RAM
> Graphics Processor: Mesa Intel® HD Graphics 4600
> Manufacturer: MSI
> Product Name: MS-7817
> System Version: 1.0

This workaround works, albeit cumbersome. Need to install wl-copy. You will lose the ability to save to file directly.

**Comment by greg2010** [comment]
Can confirm that [this](https://github.com/flameshot-org/flameshot/issues/2848#issuecomment-1199796142) patch works for me and fixes the issue of the uploaded image URL not being copied to clipboard (arch/wayland/plasma).

**Comment by AsinaMilic** [comment]
> I sketched a little script to easy build and install flameshot with `USE_WAYLAND_CLIPBOARD` flag enabled on debian based distros. It also respects new version if it will be released.
> 
> ```shell
> #!/bin/bash
> 
> set -e
> 
> rm -rf build
> mkdir -p build
> cd build
> 
> sudo apt install devscripts equivs libkf5guiaddons-dev -y
> sudo mk-build-deps -t'apt-get -y' -ir flameshot
> 
> apt source flameshot
> SRC_DIR=`ls | cut -f1 | head -n1`
> 
> pushd ${SRC_DIR}
> sed -i 's/-DFLAMESHOT_ABOUT_DEB_VERSION/-DUSE_WAYLAND_CLIPBOARD=1 -DFLAMESHOT_ABOUT_DEB_VERSION/' debian/rules
> EMAIL=nobody@email.com dch -n ''
> debuild -i -j -us -uc
> popd
> 
> sudo apt purge flameshot-build-deps devscripts equivs libkf5guiaddons-dev --auto-remove -y
> 
> sudo dpkg -i flameshot_*.deb
> rm -rf ../build
> ```

This one works for me. 
Debian 12 
Flameshot v12.1.0 (Debian 12.1.0-2.1) 
Compiled with Qt 5.15.8

**Comment by S1xZ** [comment]
> > I sketched a little script to easy build and install flameshot with `USE_WAYLAND_CLIPBOARD` flag enabled on debian based distros. It also respects new version if it will be released.
> > ```shell
> > #!/bin/bash
> > 
> > set -e
> > 
> > rm -rf build
> > mkdir -p build
> > cd build
> > 
> > sudo apt install devscripts equivs libkf5guiaddons-dev -y
> > sudo mk-build-deps -t'apt-get -y' -ir flameshot
> > 
> > apt source flameshot
> > SRC_DIR=`ls | cut -f1 | head -n1`
> > 
> > pushd ${SRC_DIR}
> > sed -i 's/-DFLAMESHOT_ABOUT_DEB_VERSION/-DUSE_WAYLAND_CLIPBOARD=1 -DFLAMESHOT_ABOUT_DEB_VERSION/' debian/rules
> > EMAIL=nobody@email.com dch -n ''
> > debuild -i -j -us -uc
> > popd
> > 
> > sudo apt purge flameshot-build-deps devscripts equivs libkf5guiaddons-dev --auto-remove -y
> > 
> > sudo dpkg -i flameshot_*.deb
> > rm -rf ../build
> > ```
> 
> This one works for me. Debian 12 Flameshot v12.1.0 (Debian 12.1.0-2.1) Compiled with Qt 5.15.8

Works like a charm! Thanks man.

**Comment by ayoubdya** [comment]
I had the same issue on Wayland of copying the URL to clipboard after uploading the screenshot to Imgur 
I fixed the issue by enabling the `USE_WAYLAND_CLIPBOARD` flag and using this [patch](https://github.com/flameshot-org/flameshot/issues/2848#issuecomment-1199796142)

OS: Arch Linux x86_64 
Kernel: 6.7.4-arch1-1 
DE: Plasma 5.27.10 
WM: kwin 
CPU: AMD Ryzen 5 5500 (12) @ 4.267GHz 
GPU: NVIDIA GeForce GTX 1660 SUPER

**Comment by bobvanderlinden** [comment]
At the moment various distros are solving the same (edgecase-y) issue by enabling the option in their packages.
Would it be a good idea to keep the option, but enable by default in flameshot?
If there are people who do not want to depend on kguiaddons or some other reason they can still disable it.

**Comment by khaled4vokalz** [comment]
> flameshot gui --raw | wl-copy

I think this is the easiest solution for now.... until they fix this in the package itself...

**Comment by ahangarha** [comment]
In Ubuntu 23.10 with Gnome, if I run `flameshot gui` in the terminal and take one screenshot, I can continue using the app as expected. I don't pass anything else.

**Comment by mmahmoudian** [comment]
@ahangarha Are you on Wayland or X11? In general, regardless of Wayland or X11, Flameshot daemon should be running in the background to retain the clipboard (I don't know the nitty-gritty of why that have been the design choice).

**Comment by ahangarha** [comment]
I am on Wayland. I didn't mention this because the title was already about the issue on Wayland. I have no idea to contribute to the discussion. I just thought that information would be helpful.

**Comment by tmccombs** [comment]
I am also experiencing this issue on Ubuntu 24.04 with sway as my wayland compositor. 

Copy with Ctrl+c or the copy icon does nothing. Everything else seems to work fine, including saving the image.

**Comment by GetPsyched** [comment]
Having the same issue here on COSMIC alpha + Wayland. Other than copying the screenshot, everything works just fine.

**Comment by joaofranciscoguarda** [comment]
I did a change in the configuration and suddenly the screenshots were saving to the clipboard.

- Check the "use fixed path for screenshots to save".
- Change prefered save file extention to "png".

![image](https://github.com/user-attachments/assets/05cd0801-abc5-4760-91be-38f84b3f4f24)

**Comment by GetPsyched** [comment]
Unfortunately, that did not work for me.

Are you using `USE_WAYLAND_GRIM` while building Flameshot? I tried with it but then `flameshot gui` wouldn't work at all since it's only for wlroot-based compositors.

**Comment by joaofranciscoguarda** [comment]
> Are you using `USE_WAYLAND_GRIM` while building Flameshot?

I downloaded it via Discover (Flatpak), provided by default by KDE, because downloading it via pacman (terminal) or building the package wasn't working at all for me either.

**Comment by tmccombs** [comment]
that doesn't work for me either

**Comment by MarkoSagadin** [comment]
> Copying to clipboard works if we run:
> flameshot gui --raw | wl-copy

I can happily say that this works for me (finally!). I am running Wayland/Hyprland on Ubuntu 24.04, flameshot version is v12.1.0, I installed it with apt.

**Comment by MfCrizz** [comment]
Can confirm that it doesn't work for me either. Kubuntu 24.04 lts, KDE 5.27.12 Wayland, Flameshot v12.1.0. 
`flameshot gui --raw | wl-copy -f` works for me from the console, but when using it as shortcut it doesn't work.

**Comment by zorn-v** [comment]
> Can confirm that it doesn't work for me either. Kubuntu 24.04 lts

Why you use it in KDE and especially with wayland ? 
Spectacle is great tool that can totally replace flameshot. 
Also it can record video in wayland.

**Comment by GetPsyched** [comment]
> Spectacle is great tool that can totally replace flameshot.

I dislike having to interact with any kind of UI when I just want to take a screenshot; the fact that it's not the default amazes me.

**Comment by zorn-v** [comment]
> I don't think you could pin a screenshot with Spectacle while flameshot could

Well, maybe not totally or not for all 😄  But I don't need that feature.

> I dislike having to interact with any kind of UI when I just want to take a screenshot; the fact that it's not the default amazes me.

There is shortcuts to any type (full, region, record etc.), what are you talking about ?

**Comment by GetPsyched** [comment]
I guess it's a matter of preference, then. I don't want the default behaviour of `PrtSc` to open up a UI and let me choose.

**Comment by zorn-v** [comment]
Press Shift+PrtSc what a problem ?
Or just bind PrtSc to "Captire Entire Desktop" (or whatever you need) in spectacle settings.

**Comment by atoav** [comment]
Also have this issue:
- Installed flameshot via apt: `Installed: 12.1.0-2build2`
- flameshot version: `Flameshot v12.1.0 (Debian 12.1.0-2build2), Compiled with Qt 5.15.12`
- Running `Kubuntu 24.04`


Output of `inxi --width 80 --system --graphics`:

```
System:
  Host: tx Kernel: 6.11.0-109019 arch: x86_64 bits: 64
  Desktop: KDE Plasma v: 6.3.2 Distro: KUBUNTU 24.04.1 LTS noble
Graphics:
  Device-1: Intel CometLake-H GT2 [UHD Graphics] driver: i915 v: kernel
  Device-2: NVIDIA TU104M [GeForce RTX 2080 SUPER Mobile / Max-Q]
    driver: nvidia v: 560.35.03
  Display: wayland server: X.org v: 1.21.1.11 with: Xwayland v: 24.1.2
    compositor: kwin_wayland driver: X: loaded: modesetting,nvidia
    unloaded: fbdev,nouveau,vesa dri: iris gpu: i915,nvidia
    resolution: 1920x1080
  API: EGL v: 1.5 drivers: iris,nvidia
    platforms: gbm,wayland,x11,surfaceless,device
  API: OpenGL v: 4.6.0 compat-v: 4.6 vendor: intel mesa
    v: 24.3.4-1~24.04 renderer: Mesa Intel UHD Graphics (CML GT2)
  API: Vulkan v: 1.3.275 drivers: N/A surfaces: xcb,xlib,wayland
```

This is on the dbus when I run the copy command

<details>
<summary>The dbus-monitor content</summary>
<pre>dbus-monitor --session sender=org.flameshot.Flameshot
signal time=1742372830.720098 sender=org.freedesktop.DBus -> destination=:1.106 serial=2 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameAcquired
   string ":1.106"
signal time=1742372830.720186 sender=org.freedesktop.DBus -> destination=:1.106 serial=4 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameLost
   string ":1.106"
method call time=1742372837.224206 sender=:1.108 -> destination=org.freedesktop.DBus serial=13 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0='org.a11y.Bus',arg1=''"
method call time=1742372837.224256 sender=:1.108 -> destination=org.freedesktop.DBus serial=14 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=NameHasOwner
   string "org.a11y.Bus"
method call time=1742372837.224395 sender=:1.108 -> destination=org.freedesktop.DBus serial=15 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0='org.a11y.Bus'"
method call time=1742372837.224463 sender=:1.108 -> destination=org.freedesktop.DBus serial=16 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=GetNameOwner
   string "org.a11y.Bus"
method call time=1742372837.224579 sender=:1.108 -> destination=org.a11y.Bus serial=17 path=/org/a11y/bus; interface=org.freedesktop.DBus.Properties; member=Get
   string "org.a11y.Status"
   string "ScreenReaderEnabled"
method call time=1742372837.224864 sender=:1.108 -> destination=org.a11y.Bus serial=18 path=/org/a11y/bus; interface=org.freedesktop.DBus.Properties; member=Get
   string "org.a11y.Status"
   string "IsEnabled"
method call time=1742372848.715160 sender=:1.108 -> destination=org.freedesktop.Notifications serial=19 path=/org/freedesktop/Notifications; interface=org.freedesktop.DBus.Introspectable; member=Introspect
method call time=1742372848.717624 sender=:1.108 -> destination=org.freedesktop.DBus serial=20 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=AddMatch
   string "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0='org.freedesktop.Notifications'"
method call time=1742372848.717707 sender=:1.108 -> destination=org.freedesktop.DBus serial=21 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=GetNameOwner
   string "org.freedesktop.Notifications"
method call time=1742372848.718595 sender=:1.108 -> destination=org.freedesktop.Notifications serial=22 path=/org/freedesktop/Notifications; interface=org.freedesktop.Notifications; member=Notify
   string "flameshot"
   uint32 0
   string "flameshot"
   string "Flameshot Info"
   string "Capture saved to clipboard."
   array [
   ]
   array [
   ]
   int32 5000
method call time=1742372848.759934 sender=:1.108 -> destination=org.freedesktop.DBus serial=23 path=/org/freedesktop/DBus; interface=org.freedesktop.DBus; member=RemoveMatch
   string "type='signal',sender='org.freedesktop.DBus',interface='org.freedesktop.DBus',member='NameOwnerChanged',arg0='org.freedesktop.Notifications'"
method return time=1742372848.786415 sender=:1.108 -> destination=:1.111 serial=24 reply_serial=22
</pre>
</details>

This is the output of the terminal running the command:
```bash
flameshot gui
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
```

As mentioned by previous posters this works:
```bash
QT_QPA_PLATFORM=wayland flameshot gui --raw | wl-copy -f
```

however so does this
```bash
flameshot gui --raw | wl-copy
```

The `--raw` flag outputs the binary data of the screenshot to stdout and pipes it into the `wl-copy` command, so it bypasses any clipboard communications flameshot would have to do. This however means that the clipboard support in wayland is still broken (as suggested by the stderr output above: `qt.qpa.wayland: Wayland does not support QWindow::requestActivate()`)

**Comment by neoscopio** [comment]
This bug is preventing me (I guess many others) to move to wayland. There are enough comments here to remove the "unconfirmed bug" label.

**Comment by michal-ciesielski-ph** [comment]
@atoav your workaround saved my day!

**Comment by andrey-aksup** [comment]
Best workaround for me (Fedora 42, it's working for now):

`sudo dnf install wl-clipboard` - for wl-copy

Next step, create a launcher: flameshot.sh
```shell
#!/bin/bash
flameshot gui --raw | wl-copy
```

And just run it with shortcuts..

Ctrl+C not working because is RAW mode, but 'ok' button working fine

**Comment by mmahmoudian** [comment]
@neoscopio 
> This bug is preventing me (I guess many others) to move to wayland. There are enough comments here to remove the "unconfirmed bug" label.

You are right, but there is a caveat: since v12.1.0 we have done a lot in Flameshot and there is a good chance that this already fixed. It would be great if you can test the newly released v13.0.0 and let us know if the issue persists

**Comment by ahangarha** [comment]
@mmahmoudian I just installed v13.0.1. I still cannot copy to clipboard.

I am on Ubuntu 24.04.2 (gnome).

**Comment by ihareg** [comment]
The same issue on Manjaro
KDE Plasma v: 6.3.6 + 
Flameshot v13.0.1 ()
Compiled with Qt 6.9.1

**Comment by ihareg** [comment]
I've checked it with v
Flameshot v13.1.0 ()
Compiled with Qt 6.9.1

If the app was triggered by shortkey and selected area was copied by CTRL+C or by clicking on Copy icon on capture screen -> Then the picture won't be pasted at all.

But if the app was triggered by clicking on the icon in a tray and after that  a Copy Icon is used on capture screen -> It's possible to install captured image to any app.

**Comment by Trevortds** [comment]
This issue has been resolved for me, either by switching to a Snap package, or upgrading to 13.0.1, or both. 

Ubuntu 24.04 
Regolith-Sway 4.9.0

Troubleshooting steps: 

- I had the issue as described with the current Debian package for some time, which I believe was 12.1.0. Could not copy to clipboard with either the button or the keyboard shortcut. The program would exit, but the screenshot and any annotations would just be lost forever because they weren't in the clipboard or on disk. 
- Uninstalled the deb
- Installed 13.0.1 from Snap latest/edge
- Now seems to be fully functional, can copy to clipboard with both the button and the shortcut

**Comment by feligato** [comment]
> Best workaround for me (Fedora 42, it's working for now):
> 
> `sudo dnf install wl-clipboard` - for wl-copy
> 
> Next step, create a launcher: flameshot.sh
> 
> #!/bin/bash
> flameshot gui --raw | wl-copy
> And just run it with shortcuts..
> 
> Ctrl+C not working because is RAW mode, but 'ok' button working fine

This worked for me too, thanks [andrey-aksup](https://github.com/andrey-aksup)

**Comment by 0cv** [comment]
Confirm this workaround with wl-copy to be working too, but this introduces now a new issue: save to disk isn't working anymore (...)

**Comment by PassionateBytes** [comment]
> After building with `cmake ../ -DUSE_WAYLAND_CLIPBOARD=true`, copying image now works on Wayland, both by Ctrl+C and by clicking button.

Does that mean this is a build-pipeline issue? Should there be a separate build for wayland platforms?

**Comment by borgmanJeremy** [comment]
I don't want to release a Wayland and non Wayland build so maybe it should be a runtime option

**Comment by tmccombs** [comment]
Could it automatically detect if it is being used in wayland by looking at the WAYLAND_DISPLAY environment variable?

**Comment by fanbox-dev** [comment]
Have the same issue after the recent software update on Fedora Linux 42 (Workstation Edition) with GNOME 48.4 DE on Wayland. 
Yesterday `flameshot gui` command copied to clipboard just fine.
The proposed `flameshot gui --raw | wl-copy -f` workaround works but now I have to click on the icon and the most important: saving screenshot to disc is not working

**Comment by eu6en** [comment]
Have the same issue after the recent software update on Fedora Linux 42 (Workstation Edition) with GNOME 48.4 DE on Wayland. 
Yesterday `flameshot gui` command copied to clipboard just fine.
The proposed `flameshot gui --raw | wl-copy -f` workaround works but now I have to click on the icon and the most important: saving screenshot to disc is not working

**Comment by Anh-V-N** [comment]
> Have the same issue after the recent software update on Fedora Linux 42 (Workstation Edition) with GNOME 48.4 DE on Wayland. Yesterday `flameshot gui` command copied to clipboard just fine. The proposed `flameshot gui --raw | wl-copy -f` workaround works but now I have to click on the icon and the most important: saving screenshot to disc is not working

you can add --path to flameshot gui to save to disc. e.g `flameshot gui --raw --path  ~/Pictures/Screenshots/screenshot-$(date +'%Y-%m-%d_%H-%M-%S').png | wl-copy -f `

**Comment by chpasha** [comment]
Hi everybody,
I have Debian 13 with KDE, I've compiled flameshot from source with USE_WAYLAND_CLIPBOARD=1 flag (I tried both the flag and CmakeLists.txt (setting the option to ON) but still cannot capture screenshots after running flameshot gui. What am I doing wrong?

**Comment by Megamannen** [comment]
Tagged "Unconfirmed" ???
Still an inssue here 3 years later

Fun workaround :P 
```
flameshot screen --raw | satty --filename - --fullscreen --save-after-copy --early-exit
```

**Comment by hunterjackson** [comment]
For those of you like me who may have found this issue recently, this is being worked on in this issue and seems to have a resolution https://github.com/flameshot-org/flameshot/issues/4116

**Comment by chpasha** [comment]
Just checked the spectacle today and found it to be not worse than flameshot for all my usage scenarios. So for those, who are on KDE, don't waste time waiting or compiling from source or whatever. I regret to not have checked it right aways after moving to Wayland, I have moved from it to flameshot long time ago for reason I don't remember any more, and now it is time to move back ;-)

**Comment by ironhak** [comment]
The issue is still there.

**Comment by ascopes** [comment]
any update to this, or any known alternatives that work sensibly on GNOME Wayland that allow annotating screenshots?

**Comment by ghost** [comment]
Copy Not working on Cosmic Desktop but works on Gnome 49.
(Compiled from Source)

**Comment by ghost** [comment]
> any update to this, or any known alternatives that work sensibly on GNOME Wayland that allow annotating screenshots?

`Gradia` works and is the closest i found.
https://gradia.alexandervanhee.be/
https://github.com/AlexanderVanhee/Gradia

**Comment by sglbl** [comment]
I just tried the [latest pre-release version](https://github.com/flameshot-org/flameshot/releases) v14.0 and the problem is solved here: (GNOME 46 - Wayland)

<img width="244" height="348" alt="Image" src="https://github.com/user-attachments/assets/807bbd0f-e29e-4b7a-b5d0-f74a5c936cec" />

Url:
[v14.0.rc1](https://github.com/flameshot-org/flameshot/releases/expanded_assets/v14.0.rc1)

In my case I selected ubuntu 24.04 based amd64.deb version. The only problem left is the flashlight and the very loud photo sound. (I guess wayland-gnome related), X11 didn't have this problem.

**Setting a shortcut:**
From Settings → Keyboard → Keyboard Shortcuts → Custom Shortcuts, Iadded this:
**Name:** `Screenshot with Flameshot`
**Command:** `/bin/sh -c '/usr/bin/flameshot gui'` (Assuming that you're not using flatpak package)
**Shortcut:** `Ctrl <` (or your shortcut)

<img width="1919" height="1078" alt="Image" src="https://github.com/user-attachments/assets/140e90ca-dec4-4b74-aedb-7d5a6ae325b2" />

**Comment by mmahmoudian** [comment]
@sglbl 
> I just tried the [latest pre-release version](https://github.com/flameshot-org/flameshot/releases) v14.0 and the problem is solved here: (GNOME 46 - Wayland)

Thanks for letting us know. This is a valuable feedback.

> In my case I selected ubuntu 24.04 based amd64.deb version. The only problem left is the flashlight and the very loud photo sound.

Yes, that's a "privacy feature" of Gnome. The idea is to inform the user that a screenshot is taken by both visual and audio notice. Perhaps you can file a feature request to make this configurable, although Gnome is not known for customizability.

**Comment by mmahmoudian** [comment]
I'll close this issue. but if anyone in this thread still have issues **after** trying [v14.0.rc1 or later](https://github.com/flameshot-org/flameshot/releases), please comment here.

**Comment by GoodMirek** [comment]
@mmahmoudian Hi Mehrad, thank you for fixing this significant issue for good!
Thank you for all your work on this project.
❤️

**Comment by sglbl** [comment]
> Yes, that's a "privacy feature" of Gnome. The idea is to inform the user that a screenshot is taken by both visual and audio notice. Perhaps you can file a feature request to make this configurable, although Gnome is not known for customizability.

Unfortunately it only happens with third party apps and solutions are only workarounds that disable other system effects/sounds as well. GNOME is known for not caring about user feedback so I doubt if they fix it soon:

https://gitlab.gnome.org/GNOME/gnome-shell/-/work_items/3866
https://gitlab.gnome.org/GNOME/gnome-shell/-/work_items/6376

**Comment by rune-scape** [comment]
i still have this issue on master with or without `USE_WAYLAND_CLIPBOARD`, no error output
```
Flameshot v15.0.0 (2d478061)
Compiled with Qt 6.8.2
```
```
        _,met$$$$$gg.
     ,g$$$$$$$$$$$$$$$P.
   ,g$$P""       """Y$$.".     OS: Debian GNU/Linux 13 (trixie) x86_64
  ,$$P'              `$$$.     Kernel: Linux 6.12.107+deb13-amd64
',$$P       ,ggs.     `$$b:    Uptime: 1 day, 8 hours, 4 mins
`d$$'     ,$P"'   .    $$$     Packages: 4330 (dpkg), 7 (flatpak-system), 39 (flatpak-user)
 $$P      d$'     ,    $$P     Shell: bash 5.2.37
 $$:      $$.   -    ,d$$'     Display (ROG PG278Q): 2560x1440 @ 144 Hz in 27" [External]
 $$;      Y$b._   _,d$P'       DE: KDE Plasma 6.3.6
 Y$$.    `.`"Y$$$$P"'          WM: KWin (Wayland)
 `$$b      "-.__               WM Theme: plastik
  `Y$$b                        Theme: Breeze (Dark) [Qt]
   `Y$$.                       Icons: breeze-dark [Qt], breeze-dark [GTK2/3/4]
     `$$b.                     Font: CaskaydiaCove Nerd Font (10pt) [Qt], CaskaydiaCove Nerd Font (10pt) [GTK2/3/4]
       `Y$$b.                  Cursor: Notwaita-Black (24px)
         `"Y$b._               Terminal: yakuake 25.4.2
             `""""             CPU: AMD Ryzen 5 3600X (12) @ 4.41 GHz
                               GPU: NVIDIA GeForce GTX 1080 Ti [Discrete]
                               Memory: 14.68 GiB / 62.73 GiB (23%)
                               Swap: 63.15 MiB / 64.00 GiB (0%)
                               Disk (/): 45.38 GiB / 59.76 GiB (76%) - ext4
                               Disk (/home): 329.81 GiB / 442.95 GiB (74%) - ext4
                               Locale: en_US.UTF-8
```
